ldk-server/src/main.rs

```diff
@@ -266,30 +294,48 @@ fn main() {
 	println!("Shutdown complete..");
 }
 
-fn upsert_payment_details(
-	event_node: &Node, paginated_store: Arc<dyn PaginatedKVStore>, payment_id: &PaymentId,
+async fn publish_event_and_upsert_payment(
+	payment_id: &PaymentId, payment_to_event: fn(Payment) -> event_envelope::Event,
+	event_node: &Node, event_publisher: Arc<dyn EventPublisher>,
+	paginated_store: Arc<dyn PaginatedKVStore>,
 ) {
 	if let Some(payment_details) = event_node.payment(payment_id) {
 		let payment = payment_to_proto(payment_details);
-		let time =
-			SystemTime::now().duration_since(UNIX_EPOCH).expect("Time must be > 1970").as_secs()
-				as i64;
-
-		match paginated_store.write(
-			PAYMENTS_PERSISTENCE_PRIMARY_NAMESPACE,
-			PAYMENTS_PERSISTENCE_SECONDARY_NAMESPACE,
-			&payment_id.0.to_lower_hex_string(),
-			time,
-			&payment.encode_to_vec(),
-		) {
-			Ok(_) => {
-				event_node.event_handled();
-			},
+
+		let event = payment_to_event(payment.clone());
+		let event_name = get_event_name(&event);
+		match event_publisher.publish(EventEnvelope { event: Some(event) }).await {
+			Ok(_) => {},
 			Err(e) => {
-				eprintln!("Failed to write payment to persistence: {}", e);
+				println!("Failed to publish '{}' event: {}", event_name, e);
```

Comment:
Would it be worth providing more information other than the event name? Does it alone even give us any useful information? Maybe include the payment id?